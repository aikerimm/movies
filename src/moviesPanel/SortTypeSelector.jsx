import React from 'react';
import PropTypes from 'prop-types';

const SortTypeSelector = ({ sortType, sortDirection, onSortChanged }) => {
  return (
    <div className='sortTypeSelect'>
      <label htmlFor='sortType' className='sortTypeLabel'>
        Sort by
      </label>
      <select
        name='sortType'
        id='sortType'
        onChange={({ target: { value } }) =>
          onSortChanged(value, sortDirection)
        }
        value={sortType || ''}
      >
        <option value=''>Select: </option>
        <a href={onSortChanged('release_date', sortDirection)}>
          {' '}
          <option value='release_date'>Release Date</option>
        </a>
        <a href={onSortChanged('title', sortDirection)}>
          <option value='title'>Title</option>
        </a>
        <a href={onSortChanged('vote_average', sortDirection)}>
          <option value='vote_average'>Rating</option>
        </a>
      </select>
      <a
        href={onSortChanged(
          sortType ? sortType : 'release_date',
          sortDirection === 'asc' ? 'desc' : 'asc'
        )}
      >
        <div>
          <span className='sortDirection'>
            {sortDirection === 'asc' ? '⇑' : '⇓'}
          </span>
        </div>
      </a>
    </div>
  );
};

SortTypeSelector.propTypes = {
  sortType: PropTypes.string,
  sortDirection: PropTypes.string.isRequired,
  onSortChanged: PropTypes.func.isRequired,
};

export default React.memo(SortTypeSelector);
