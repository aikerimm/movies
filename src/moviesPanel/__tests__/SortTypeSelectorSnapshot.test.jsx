import renderer from 'react-test-renderer';
import SortTypeSelector from '../SortTypeSelector';

it('SortTypeSelector should render correctly', () => {
  const tree = renderer
    .create(
      <SortTypeSelector
        sortType='TITLE'
        sortDirection='asc'
        onSortChanged={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
