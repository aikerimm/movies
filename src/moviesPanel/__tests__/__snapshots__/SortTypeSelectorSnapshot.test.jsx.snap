// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortTypeSelector should render correctly 1`] = `
<div
  className="sortTypeSelect"
>
  <label
    className="sortTypeLabel"
    htmlFor="sortType"
  >
    Sort by
  </label>
  <select
    id="sortType"
    name="sortType"
    onChange={[Function]}
    value="TITLE"
  >
    <option
      value=""
    >
      Select: 
    </option>
    <option
      value="release_date"
    >
      Release Date
    </option>
    <option
      value="title"
    >
      Title
    </option>
    <option
      value="vote_average"
    >
      Rating
    </option>
  </select>
  <div
    onClick={[Function]}
  >
    <span
      className="sortDirection"
    >
      â‡‘
    </span>
  </div>
</div>
`;
